src/solver/impls/rkgeneric/rkgeneric.cxx

```diff
@@ -78,6 +79,16 @@ int RKGenericSolver::init(int nout, BoutReal tstep) {
   return 0;
 }
 
+void RKGenericSolver::resetInternalFields(){
+  //Zero out history
+  for(int i=0;i<nlocal;i++){
```

Comment:
Could this be a range-based loop?

Reply:
It could be replaced with

~~~~
tmpState.assign(nlocal, 0); // or tmpState.assign(tmpState.size(),0);
f2.assign(nlocal, 0); // or f2.assign(f2.size(),0);
~~~~

Reply:
I was just about to suggest `std::fill`, but `assign` with `size` looks better

Reply:
Ah oops I thought these were vectors.